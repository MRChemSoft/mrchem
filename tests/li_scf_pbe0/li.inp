# vim:syntax=sh:

world_prec = 1.0e-3               # Overall relative precision
world_size = 5                    # Size of simulation box 2^n

MPI {
  numerically_exact = true        # Guarantee identical results in MPI
}

Molecule {
multiplicity = 2
$coords
Li   0.0     0.0      0.0
$end
}

WaveFunction {
  method = PBE0                   # Wave function method (HF or DFT)
  restricted = false
}

SCF {
  kain = 3                        # Length of KAIN iterative history
  max_iter = 5
  orbital_thrs = 2.0e-2
  guess_type = GTO             # Type of initial guess (none, gto, mw)
}
