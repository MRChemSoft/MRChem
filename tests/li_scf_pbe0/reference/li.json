{
  "input": {
    "molecule": {
      "cavity_coords": [
        {
          "center": [
            0.0,
            0.0,
            0.0
          ],
          "radius": 2.05
        }
      ],
      "cavity_width": 0.2,
      "charge": 0,
      "coords": [
        {
          "atom": "li",
          "xyz": [
            0.0,
            0.0,
            0.0
          ]
        }
      ],
      "multiplicity": 2
    },
    "mpi": {
      "bank_size": -1,
      "numerically_exact": true,
      "shared_memory_size": 10000
    },
    "mra": {
      "basis_order": 5,
      "basis_type": "interpolating",
      "boxes": [
        2,
        2,
        2
      ],
      "corner": [
        -1,
        -1,
        -1
      ],
      "max_scale": 20,
      "min_scale": -4
    },
    "printer": {
      "file_name": "li.inp",
      "print_level": 0,
      "print_mpi": false,
      "print_prec": 6,
      "print_width": 75
    },
    "rsp_calculations": {},
    "scf_calculation": {
      "fock_operator": {
        "coulomb_operator": {
          "poisson_prec": 0.001,
          "shared_memory": false
        },
        "exchange_operator": {
          "exchange_prec": -1.0,
          "poisson_prec": 0.001
        },
        "kinetic_operator": {
          "derivative": "abgv_55"
        },
        "nuclear_operator": {
          "proj_prec": 0.001,
          "shared_memory": false,
          "smooth_prec": 0.001
        },
        "xc_operator": {
          "shared_memory": false,
          "xc_functional": {
            "cutoff": 0.0,
            "functionals": [
              {
                "coef": 1.0,
                "name": "pbe0"
              }
            ],
            "spin": true
          }
        }
      },
      "initial_guess": {
        "file_basis": "initial_guess/mrchem.bas",
        "file_chk": "checkpoint/phi_scf",
        "file_gto_a": "initial_guess/mrchem.moa",
        "file_gto_b": "initial_guess/mrchem.mob",
        "file_gto_p": "initial_guess/mrchem.mop",
        "file_phi_a": "initial_guess/phi_a_scf",
        "file_phi_b": "initial_guess/phi_b_scf",
        "file_phi_p": "initial_guess/phi_p_scf",
        "localize": false,
        "method": "DFT (PBE0)",
        "prec": 0.001,
        "restricted": false,
        "type": "gto",
        "zeta": 0
      },
      "properties": {
        "dipole_moment": {
          "dip-1": {
            "operator": "h_e_dip",
            "precision": 0.001,
            "r_O": [
              0.0,
              0.0,
              0.0
            ]
          }
        }
      },
      "scf_solver": {
        "checkpoint": false,
        "energy_thrs": -1.0,
        "file_chk": "checkpoint/phi_scf",
        "final_prec": 0.001,
        "helmholtz_prec": -1.0,
        "kain": 3,
        "localize": false,
        "max_iter": 5,
        "method": "DFT (PBE0)",
        "orbital_thrs": 0.02,
        "rotation": 0,
        "start_prec": 0.001
      }
    },
    "schema_name": "mrchem_input",
    "schema_version": 1
  },
  "output": {
    "properties": {
      "center_of_mass": [
        0.0,
        0.0,
        0.0
      ],
      "charge": 0,
      "dipole_moment": {
        "dip-1": {
          "magnitude": 6.707146772090342e-14,
          "r_O": [
            0.0,
            0.0,
            0.0
          ],
          "vector": [
            0.0,
            0.0,
            0.0
          ],
          "vector_el": [
            0.0,
            0.0,
            0.0
          ],
          "vector_nuc": [
            0.0,
            0.0,
            0.0
          ]
        }
      },
      "geometry": [
        {
          "symbol": "Li",
          "xyz": [
            0.0,
            0.0,
            0.0
          ]
        }
      ],
      "multiplicity": 2,
      "orbital_energies": {
        "energy": [
          -2.0558230450949067,
          -0.14105015483172156,
          -2.0455375909446403
        ],
        "occupation": [
          1.0,
          1.0,
          1.0
        ],
        "spin": [
          "a",
          "a",
          "b"
        ],
        "sum_occupied": -4.2424107908712685
      },
      "scf_energy": {
        "E_ee": 4.048635375787536,
        "E_eext": 0.0,
        "E_el": -7.465980283128326,
        "E_en": -17.08879205743869,
        "E_kin": 7.38046968695824,
        "E_next": 0.0,
        "E_nn": 0.0,
        "E_nuc": 0.0,
        "E_tot": -7.465980283128326,
        "E_x": -0.44272368481967017,
        "E_xc": -1.3635696036157432,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      }
    },
    "provenance": {
      "creator": "MRChem",
      "mpi_processes": 1,
      "nthreads": 4,
      "routine": "mrchem.x",
      "total_cores": 4,
      "version": "1.1.0-alpha"
    },
    "rsp_calculations": null,
    "scf_calculation": {
      "initial_energy": {
        "E_ee": 4.061917141108662,
        "E_eext": 0.0,
        "E_el": -7.41376983264562,
        "E_en": -17.03503042962955,
        "E_kin": 7.370996808590605,
        "E_next": 0.0,
        "E_nn": 0.0,
        "E_nuc": 0.0,
        "E_tot": -7.41376983264562,
        "E_x": -0.44476857209247533,
        "E_xc": -1.3668847806228603,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      },
      "scf_solver": {
        "converged": true,
        "cycles": [
          {
            "energy_terms": {
              "E_ee": 4.035331828637668,
              "E_eext": 0.0,
              "E_el": -7.465151310507911,
              "E_en": -17.01194236139491,
              "E_kin": 7.311740389767914,
              "E_next": 0.0,
              "E_nn": 0.0,
              "E_nuc": 0.0,
              "E_tot": -7.465151310507911,
              "E_x": -0.44138486093815277,
              "E_xc": -1.3588963065804285,
              "Er_el": 0.0,
              "Er_nuc": 0.0,
              "Er_tot": 0.0
            },
            "energy_total": -7.465151310507911,
            "energy_update": 0.0513814778622903,
            "mo_residual": 0.06988079295538935,
            "wall_time": 5.610890154
          },
          {
            "energy_terms": {
              "E_ee": 4.048635375787536,
              "E_eext": 0.0,
              "E_el": -7.465980283128326,
              "E_en": -17.08879205743869,
              "E_kin": 7.38046968695824,
              "E_next": 0.0,
              "E_nn": 0.0,
              "E_nuc": 0.0,
              "E_tot": -7.465980283128326,
              "E_x": -0.44272368481967017,
              "E_xc": -1.3635696036157432,
              "Er_el": 0.0,
              "Er_nuc": 0.0,
              "Er_tot": 0.0
            },
            "energy_total": -7.465980283128326,
            "energy_update": 0.0008289726204155912,
            "mo_residual": 0.01755830846088095,
            "wall_time": 4.132942602
          }
        ],
        "wall_time": 9.744296584
      },
      "success": true
    },
    "schema_name": "mrchem_output",
    "schema_version": 1,
    "success": true
  }
}
