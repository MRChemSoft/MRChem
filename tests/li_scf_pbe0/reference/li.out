                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                                                                     ***
***          __  __ ____   ____ _                                       ***
***         |  \/  |  _ \ / ___| |__   ___ _ __ ___                     ***
***         | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                    ***
***         | |  | |  _ <| |___| | | |  __/ | | | | |                   ***
***         |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                   ***
***                                                                     ***
***         VERSION            1.1.0-alpha                              ***
***                                                                     ***
***         Git branch         bugfix/exchange                          ***
***         Git commit hash    294d81970e568e38d589                     ***
***         Git commit author  Stig Rune Jensen                         ***
***         Git commit date    Thu Jul 8 13:47:28 2021 +0200            ***
***                                                                     ***
***         Contact: luca.frediani@uit.no                               ***
***                                                                     ***
***         Radovan Bast            Magnar Bjorgve                      ***
***         Roberto Di Remigio      Antoine Durdek                      ***
***         Luca Frediani           Gabriel Gerez                       ***
***         Stig Rune Jensen        Jonas Juselius                      ***
***         Rune Monstad            Peter Wind                          ***
***                                                                     ***
***************************************************************************

---------------------------------------------------------------------------

 MPI processes           :       (no bank)                               1
 OpenMP threads          :                                               4
 Total cores             :                                               4
                                                                           
---------------------------------------------------------------------------

XCFun DFT library Copyright 2009-2020 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

---------------------------------------------------------------------------

 MRCPP version         : 1.4.0-alpha
 Git branch            : HEAD
 Git commit hash       : 2797a7a18efb69942a6e
 Git commit author     : Stig Rune Jensen
 Git commit date       : Fri Mar 12 12:48:07 2021 +0100

 Linear algebra        : EIGEN v3.3.7
 Parallelization       : OpenMP (4 threads)

---------------------------------------------------------------------------


                                                                           
===========================================================================
                         MultiResolution Analysis
---------------------------------------------------------------------------
 polynomial order      : 5
 polynomial type       : Interpolating
---------------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   16.00000    16.00000    16.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -16.00000   -16.00000   -16.00000 ]
 upper bounds          : [   16.00000    16.00000    16.00000 ]
 total length          : [   32.00000    32.00000    32.00000 ]
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                        Initializing Molecule                        ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               2
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0      Li            :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.000000
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                 Computing Ground State Wavefunction                 ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial orbitals
 Method                  : Project GTO molecular orbitals
 Precision               : 1.00000e-03
 Restricted              : False
 MO alpha file           : initial_guess/mrchem.moa
 MO beta file            : initial_guess/mrchem.mob
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                            Molecular Orbitals
---------------------------------------------------------------------------
 Alpha electrons         :                                               2
 Beta electrons          :                                               1
 Total electrons         :                                               3
---------------------------------------------------------------------------
    n  Occ Spin          :                                            Norm
---------------------------------------------------------------------------
    0    1    a          :                              9.999999988178e-01
    1    1    a          :                              1.000000048122e+00
    2    1    b          :                              9.999999988146e-01
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial energy
 Method                  : DFT (PBE0)
 Precision               : 1.00000e-03
 Localization            : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                        Molecular Energy (initial)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  7.370996808591
 E-N energy              :            (au)                -17.035030429630
 Coulomb energy          :            (au)                  4.061917141109
 Exchange energy         :            (au)                 -0.444768572092
 X-C energy              :            (au)                 -1.366884780623
 Ext. field (el)         :            (au)                  0.000000000000
---------------------------------------------------------------------------
 N-N energy              :            (au)                  0.000000000000
 Ext. field (nuc)        :            (au)                  0.000000000000
 Tot. Reac. Energy       :            (au)                  0.000000000000
 El. Reac. Energy        :            (au)                  0.000000000000
 Nuc. Reac. Energy       :            (au)                  0.000000000000
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -7.413769832646
 Nuclear energy          :            (au)                  0.000000000000
---------------------------------------------------------------------------
 Total energy            :            (au)             -7.413769832646e+00
                         :      (kcal/mol)             -4.652210770972e+03
                         :        (kJ/mol)             -1.946484987838e+04
                         :            (eV)             -2.017389455882e+02
===========================================================================


===========================================================================
                        Orbital Energies (initial)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    1    a          :            (au)                 -2.026209005677
    1    1    a          :            (au)                 -0.139783592232
    2    1    b          :            (au)                 -2.015925200363
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.181917798272
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Optimize ground state orbitals
 Method                  : DFT (PBE0)
 Checkpointing           : Off
 Max iterations          : 5
 KAIN solver             : 3
 Localization            : Off
 Diagonalization         : First two iterations
 Start precision         : 1.00000e-03
 Final precision         : 1.00000e-03
 Helmholtz precision     : Dynamic
 Energy threshold        : Off
 Orbital threshold       : 2.00000e-02
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
 Iter           MO residual             Total energy                Update
---------------------------------------------------------------------------
    0          1.732051e+00          -7.413769832646         -7.413770e+00
    1          6.988079e-02          -7.465151310508         -5.138148e-02
    2          1.755831e-02          -7.465980283128         -8.289726e-04
---------------------------------------------------------------------------
                      SCF converged in 2 iterations!
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                    Printing Molecular Properties                    ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               2
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0      Li            :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.000000
===========================================================================


===========================================================================
                         Molecular Energy (final)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  7.380469686958
 E-N energy              :            (au)                -17.088792057439
 Coulomb energy          :            (au)                  4.048635375788
 Exchange energy         :            (au)                 -0.442723684820
 X-C energy              :            (au)                 -1.363569603616
 Ext. field (el)         :            (au)                  0.000000000000
---------------------------------------------------------------------------
 N-N energy              :            (au)                  0.000000000000
 Ext. field (nuc)        :            (au)                  0.000000000000
 Tot. Reac. Energy       :            (au)                  0.000000000000
 El. Reac. Energy        :            (au)                  0.000000000000
 Nuc. Reac. Energy       :            (au)                  0.000000000000
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -7.465980283128
 Nuclear energy          :            (au)                  0.000000000000
---------------------------------------------------------------------------
 Total energy            :            (au)             -7.465980283128e+00
                         :      (kcal/mol)             -4.684973323030e+03
                         :        (kJ/mol)             -1.960192839628e+04
                         :            (eV)             -2.031596642599e+02
===========================================================================


===========================================================================
                         Orbital Energies (final)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    1    a          :            (au)                 -2.055823045095
    1    1    a          :            (au)                 -0.141050154832
    2    1    b          :            (au)                 -2.045537590945
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.242410790871
===========================================================================


===========================================================================
                           Dipole Moment (dip-1)
---------------------------------------------------------------------------
 r_O                     :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Electronic vector       :       -0.000000       -0.000000       -0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
---------------------------------------------------------------------------
 Nuclear vector          :        0.000000        0.000000        0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
---------------------------------------------------------------------------
 Total vector            :        0.000000        0.000000        0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
===========================================================================


                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                            Exiting MRChem                           ***
***                                                                     ***
***                       Wall time :  0h  0m 12s                       ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
